statistics/handle/update.go

```diff
@@ -954,6 +954,22 @@ func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) (analyzed bool) {
 		return false
 	}
 	pruneMode := h.CurrentPruneMode()
+	var dbAnalyzed string
+	var tblAnalyzed *model.TableInfo
+	hasAnalyzed := false
+	defer func() {
+		historicalStatsEnabled, err := h.CheckHistoricalStatsEnable()
+		if err != nil {
+			logutil.BgLogger().Error("Check TiDBEnableHistoricalStats Failed.")
+		}
+		if historicalStatsEnabled && hasAnalyzed {
+			if _, err := h.SaveHistoryStatsToStorage(dbAnalyzed, tblAnalyzed); err != nil {
+				logutil.BgLogger().Error("Save historical stats failed.", zap.String("db", dbAnalyzed),
+					zap.String("table", tblAnalyzed.Name.O))
+			}
+		}
+	}()
```

Comment:
The implementation of `auto-analyze` is creating an analyzing SQL `analyze ...` and processing it.
So it's better to move this logic to the end of `analyze`(`AnalyzeExec.Next`) and then `analyze` can be handled as well.

Reply:
Better choice. I' ll move the historical stats dump logic to Next directly.